### PR TITLE
CNV-36070: Storageclass migration select volumes

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Add the select volume step for the migration. 

## 🎥 Demo

https://github.com/user-attachments/assets/54af681b-489f-4889-9618-492b126ca790

**Fix printable disk drive**

Screenshot 2024-11-04 at 17 15 30

### CODE REVIEW

File: src/views/virtualmachines/migrate/tabs/VirtualMachineMigrationDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const { t } = useKubevirtTranslation();
+
+  const allVolumes = entireVMSelected(selectedPVCs);
+
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <Title headingLevel="h2">{t('Migration details')}</Title>
+        <Text component={TextVariants.p}>
+          <Trans t={t}>
```
Comment (avivtur) [inline src/views/virtualmachines/migrate/tabs/VirtualMachineMigrationDetails.tsx:38]: Is Trans really needed here? can you just translate the text?

File: src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { getTableDiskData } from './utils';
+
+type DiskTableProps = {
+  pvcs: IoK8sApiCoreV1PersistentVolumeClaim[];
+  selectedPVCs: IoK8sApiCoreV1PersistentVolumeClaim[];
+  setSelectedPVCs: Dispatch<SetStateAction<IoK8sApiCoreV1PersistentVolumeClaim[]>>;
+  vm: V1VirtualMachine;
+};
+
+const columnNames = {
```
Comment (avivtur) [inline src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx:19]: move to constants file please

File: src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  drive: 'Drive',
+  name: 'Name',
+  size: 'Size',
+  storageClass: 'Storage class',
+};
+
+const DiskTable: FC<DiskTableProps> = ({ pvcs, selectedPVCs, setSelectedPVCs, vm }) => {
```
Comment (avivtur) [inline src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx:26]: I think we already have a component called diskTable or something very similar in our VM details view/wizard view, can you rename this component?

File: src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    <Table aria-label="Selectable table">
+      <Thead>
+        <Tr>
+          <Th
+            select={{
+              isSelected: selectedPVCs?.length === selectableData.length,
+              onSelect: (_event, isSelecting) =>
+                isSelecting
+                  ? setSelectedPVCs(selectableData.map((data) => data.pvc))
+                  : setSelectedPVCs([]),
```
Comment (metalice) [inline src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx:41]: setSelectedPVCs(isSelecting ? selectableData.map((data) => data.pvc) || [])

File: src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx (hunk 3 above)
Comment (upalatucci) [reply src/views/virtualmachines/migrate/tabs/components/DiskTable.tsx:26]: renamed `SelectMigrationDisksTable`

**Pull request merged**